### PR TITLE
fix: Dynamically generate robots.txt based on Vercel environment

### PR DESCRIPTION
**Pull request by soniaklimas**
I want to merge this change because it dynamically generates robots.txt based on Vercel environment (production, preview or development).

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/next.config.js [hunk 1]
```diff
@@ -84,25 +84,6 @@ const nextConfig = withNextIntl({
   },
   reactStrictMode: true,
   transpilePackages: ["@nimara/ui"],
-  async headers() {
```
Comment (karolkarolka) [inline apps/storefront/next.config.js:87]: you should leave this code as described [here](https://vercel.com/guides/are-vercel-preview-deployment-indexed-by-search-engines) but you need to update the ENV as our custom env `NEXT_PUBLIC_ENVIRONMENT` probably haven't had any affect after being deployed to the custom domain for our Previews DEV and STAGE envs

**Review by karolkarolka — CHANGES_REQUESTED** [review]

File: apps/storefront/next.config.js [hunk 2]
```diff
@@ -86,10 +86,7 @@ const nextConfig = withNextIntl({
   transpilePackages: ["@nimara/ui"],
   async headers() {
     const headers = [];
-    if (
-      process.env.NEXT_PUBLIC_ENVIRONMENT === "DEVELOPMENT" ||
-      process.env.NEXT_PUBLIC_ENVIRONMENT === "STAGING"
-    ) {
+    if (process.env.VERCEL_ENV === "preview") {
```
Comment (karolkarolka) [inline apps/storefront/next.config.js:89]:
```suggestion
    if (process.env.VERCEL_ENV !== "production") {
```

**Review by karolkarolka — APPROVED** [review]
